cmd/common-main.go

```diff
@@ -402,6 +402,29 @@ func buildServerCtxt(ctx *cli.Context, ctxt *serverCtxt) (err error) {
 		ctxt.certsDirSet = true
 	}
 
+	memAvailable := availableMemory()
+	if ctx.IsSet("memlimit") {
+		memlimit := ctx.String("memlimit")
+		if memlimit == "" {
+			memlimit = ctx.GlobalString("memlimit")
+		}
+		mlimit, err := humanize.ParseBytes(memlimit)
+		if err != nil {
+			return err
+		}
+		if mlimit > memAvailable {
+			logger.Info("WARNING: maximum memory available (%s) smaller than specified --memlimit=%s, ignoring --memlimit value",
+				humanize.IBytes(memAvailable), memlimit)
+		}
+		ctxt.MemLimit = mlimit
+	} else {
+		ctxt.MemLimit = memAvailable
+	}
+
+	if memAvailable < ctxt.MemLimit {
+		ctxt.MemLimit = memAvailable
+	}
```

Comment:
I think let's avoid setting any limit here if --memlimit is not specified, that is the old behavior anyway. If I am not wrong, this will make the GC works very hard even before the memory usage reaches 75%

Reply:
If you are using that much memory you better GC @vadmeste instead of crashing. There is nothing wrong in setting this value.

Reply:
we should really use this value properly/appropriately because otherwise we risk of OOM in unexpected ways.